app/components/search/Refinements/ClearSearchButton.tsx

```diff
@@ -1,16 +1,30 @@
 import React from "react";
 import { Button } from "components/ui/inline/Button/Button";
-import { useSearchBox } from "react-instantsearch-core";
+import { useInstantSearch } from "react-instantsearch-core";
 
 const ClearSearchButton = () => {
-  const { clear } = useSearchBox();
+  const { setIndexUiState } = useInstantSearch();
+
+  // Algolia does provide a hook that can manage the query state, specifically *clearing*:
+  // ```
+  // const { clear } = useSearchBox();
+  // onClick(() => clear);
+  // ```
+  // However in practice using this hook results in unexpected behavior in which it ends causing
+  // unnecessary re-renders when no results are returned. Fortunately resetting the index state
+  // achieves what we need.
```

Comment:
```suggestion
  // Algolia provides a hook that can manage the query state, specifically *clearing*:
  // ```
  // const { clear } = useSearchBox();
  // onClick(() => clear);
  // ```
  // However, in practice using this hook produces unexpected behavior  causing
  // unnecessary re-renders when no results are returned. Fortunately resetting the index state
  // achieves what we need.
```